packages/flutter_image/lib/network.dart

```diff
@@ -53,6 +54,21 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
   /// fetch attempts failed.
   final FetchStrategy fetchStrategy;
 
+  /// HTTP Headers to add to the request.
+  ///
+  /// These headers will override any existing headers used by the network
+  /// library. By default all headers are changed to lowercase before sending.
+  /// To preserve the case of the headers set [preserveHeaderCase] to true
+  final Map<String, Object>? requestHeaders;
```

Comment:
Why is this a map rather than `HttpHeaders`?

Reply:
HttpClientRequest only exposes a getter for headers. And HttpHeaders and contains quite a bit [more than just the headers](https://github.com/dart-lang/sdk/blob/master/sdk/lib/_http/http_headers.dart). Wholesale overriding or copying the values one by one looks error prone. In decided to go for simplicity.